src/runtime/auto-prompt-manager.js

```diff
@@ -479,6 +479,19 @@ export class AutoPromptManager {
       return Promise.resolve();
     }
 
+    // Last impressions (which should be the mini prompt impression) should be
+    // removed from the storage after the user clicks the mini prompt. This is
+    // to prevent the impressions of both the mini and large promprts from
+    // being conunted towards the cap.
+    if (
```

Comment:
Hmmm so the flow will be:
- show miniprompt (record impression)
- user clicks mini prompt (remove last impression, new window adds impression)

Would it make more sense to instead only record a single impression? i.e. just track a bool indicating that we've already recorded an impression and no-op if we have?

```
class AutoPrompt {
  hasRecordedImpression = false;
  
  handleClientEvent_(event) {
     if (this.hasRecordedImpression) { return Promise.resolve; }
     
     if (impressionEvents.includes(event.eventType)) {
      this.hasRecordedImpression = true;
      return this.storeEvent_(STORAGE_KEY_IMPRESSIONS);
    }
  }
}
```

The only downside being that the other prompt may not come up for a period of time after but I imagine that should be ok? What do you think? This way we don't have to keep going back and forth to localstorage

Reply:
Synced offline. The only downside I can think of for this approach is that it will not work if the publisher site is a single page app. However, the current auto prompt will probably not work at all in SPA so maybe we can assume the limitation.

I made a small change of your idea by moving the `this.hasRecordedImpression` condition check inside `impressionEvents.includes(event.eventType)`. Otherwise, it would not store the dismissal.